### PR TITLE
Added new role management features

### PR DESCRIPTION
**Pull request by aleksvagachev**
##### SUMMARY
Added new role management features

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: tests/integration/targets/clickhouse_role/tasks/initial.yml [hunk 1]
```diff
@@ -33,12 +33,15 @@
...
+    settings:
+      - max_memory_usage = 15000 READONLY
+      - max_memory_usage_for_all_queries = 15000 MIN 15000 MAX 16000 WRITABLE
 
 - name: Test 2 - Check the return values
   ansible.builtin.assert:
     that:
     - result is changed
-    - result.executed_statements == ['CREATE ROLE test_role']
+    - result.executed_statements != []
```
Comment (Andersson007) [inline tests/integration/targets/clickhouse_role/tasks/initial.yml:44]:
```suggestion
    - result.executed_statements != []
```
is it an issue to put the executed query here?

**Review by Andersson007 — COMMENTED** [review]
@aleksvagachev thanks for the contribution, we should also think about making these options idempotent if possible as well as for _user module as we discussed

**Comment by aleksvagachev** [comment]
@Andersson007 Will we do idenpotency in this release or in the next one?

File: tests/integration/targets/clickhouse_role/tasks/initial.yml (hunk 1 above)
Comment (aleksvagachev) [reply tests/integration/targets/clickhouse_role/tasks/initial.yml:44]: The query will be long, I decided not to specify it.

**Comment by Andersson007** [comment]
> @Andersson007 Will we do idenpotency in this release or in the next one?

@aleksvagachev i don't know, as you prefer. If i had capacity to add idempotency myself very soon, i would wait but i don't have the capacity. Anyway, to release it now, SGTM

File: tests/integration/targets/clickhouse_role/tasks/initial.yml (hunk 1 above)
Comment (oraNod) [reply tests/integration/targets/clickhouse_role/tasks/initial.yml:44]: +1 to including the query no matter the length

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@aleksvagachev thanks for the contribution! @oraNod thanks for reviewing!

**Pull request merged**